test/built-ins/Temporal/Calendar/prototype/monthDayFromFields/missing-properties.js

```diff
@@ -23,20 +23,21 @@ const missingDay = {
 };
 assert.throws(TypeError, () => instance.monthDayFromFields(missingDay), "day should be checked before year and month");
 
-let getMonthCode = false;
-let getYear = false;
-const monthWithoutYear = {
-  day: 1,
-  month: 5,
+let got = [];
+const fieldsSpy = {
+  get year() {
+    got.push("year");
+  },
+  get month() {
+    got.push("month");
+  },
   get monthCode() {
-    getMonthCode = true;
+    got.push("monthCode");
   },
-  get year() {
-    getYear = true;
+  get day() {
+    got.push("day");
+    return 1;
   },
 };
```

Comment:
You could use TemporalHelpers here:
```js
const fieldsSpy = TemporalHelpers.propertyBagObserver(got, {
  day: 1,
});
```

Reply:
Aside from `propertyBagObserver` seeming misplaced on `TemporalHelpers`, that seems fine. I have no strong feelings here.

Reply:
Yes, at some point I'd like to move it into its own helper file that's not named Temporal.